cpp/tests/hashing/hash_test.cu

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cudf/hashing.hpp>
+#include <tests/utilities/base_fixture.hpp>
+#include <tests/utilities/type_lists.hpp>
+#include <tests/utilities/column_utilities.hpp>
+#include <tests/utilities/column_wrapper.hpp>
+
+// for columns_equal
+#include <cudf/table/table_device_view.cuh>
+#include <cudf/table/row_operators.cuh>
+
+namespace {
+
+// TODO move these to column_utilities?
+
+#define RETURN_FALSE_IF_NEQ(a, b) if (not ((a) == (b))) { return false; }
+
+bool column_properties_equal(cudf::column_view lhs, cudf::column_view rhs) {
+  RETURN_FALSE_IF_NEQ(lhs.type(), rhs.type());
+  RETURN_FALSE_IF_NEQ(lhs.size(), rhs.size());
+  RETURN_FALSE_IF_NEQ(lhs.null_count(), rhs.null_count());
+  if (lhs.size() > 0) {
+    RETURN_FALSE_IF_NEQ(lhs.nullable(), rhs.nullable());
+  }
+  RETURN_FALSE_IF_NEQ(lhs.has_nulls(), rhs.has_nulls());
+  RETURN_FALSE_IF_NEQ(lhs.num_children(), rhs.num_children());
+  return true;
+}
+
+bool columns_equal(cudf::column_view lhs, cudf::column_view rhs) {
+  using cudf::experimental::row_equality_comparator;
+
+  if (not column_properties_equal(lhs, rhs))
+    return false;
+
+  auto d_lhs = cudf::table_device_view::create(cudf::table_view{{lhs}});
+  auto d_rhs = cudf::table_device_view::create(cudf::table_view{{rhs}});
+
+  return thrust::equal(thrust::device, thrust::make_counting_iterator(0),
+                       thrust::make_counting_iterator(lhs.size()),
+                       thrust::make_counting_iterator(0),
+                       row_equality_comparator<true>{*d_lhs, *d_rhs});
+}
+
+}
+
+class HashTest : public cudf::test::BaseFixture {};
+
+TEST_F(HashTest, MultiValue)
+{
+  using cudf::test::fixed_width_column_wrapper;
+  using cudf::test::strings_column_wrapper;
+  using cudf::experimental::bool8;
+
+  auto const strings_col = strings_column_wrapper(
```

Comment:
This line is giving you a compilation error on ubuntu16.04-cuda9.2

```suggestion
  strings_column_wrapper strings_col({
```

Reply:
Good catch! I had this same problem in my scatter tests as well. Seems to be a gcc bug :(